rct229/rules/section21/section21rule10.py

```diff
@@ -0,0 +1,112 @@
+    HVAC_SYS.SYS_5,
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_11_2,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_1A,
+    HVAC_SYS.SYS_7A,
+    HVAC_SYS.SYS_11_2A,
+    HVAC_SYS.SYS_12A,
+    HVAC_SYS.SYS_1B,
+    HVAC_SYS.SYS_3B,
+    HVAC_SYS.SYS_5B,
+    HVAC_SYS.SYS_6B,
+    HVAC_SYS.SYS_7B,
+    HVAC_SYS.SYS_8B,
+    HVAC_SYS.SYS_9B,
+    HVAC_SYS.SYS_11_1B,
+    HVAC_SYS.SYS_12B,
+    HVAC_SYS.SYS_1C,
+    HVAC_SYS.SYS_3C,
+    HVAC_SYS.SYS_7C,
+    HVAC_SYS.SYS_11_1C,
+    HVAC_SYS.SYS_12C,
+]
+
+PUMP_SPEED_CONTROL = schema_enums["PumpSpeedControlOptions"]
+PUMP_CONFIGURATION_THRESHOLD = 12000 * ureg("ft2")
+
+
+class Section21Rule10(RuleDefinitionListIndexedBase):
+    """Rule 10 of ASHRAE 90.1-2019 Appendix G Section 23 (Hot water loop)"""
+
+    def __init__(self):
+        super(Section21Rule10, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section21Rule10.PumpRule(),
+            index_rmr="baseline",
+            id="21-10",
+            description="When the building is modeled with HHW plant (served by either boiler(s) or purchased hot "
+            "water/steam), the hot water pump shall be modeled as riding the pump curve if the hot water "
+            "system serves less than 120,000 ft^2 otherwise it shall be modeled with a VFD.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="pumps[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list contains all HVAC systems that are modeled in the rmi_b
+        available_type_lists = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+        return any(
+            [
+                available_type in APPLICABLE_SYS_TYPES
+                for available_type in available_type_lists
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        # to avoid pumps in service water heating system
+        loop_zone_list_w_area_dict = get_hw_loop_zone_list_w_area(rmi_b)
+
+        return {"loop_zone_list_w_area_dict": loop_zone_list_w_area_dict}
+
+    def list_filter(self, context_item, data):
+        pump = context_item.baseline
+        loop_or_piping = getattr_(pump, "Pump", "loop_or_piping")
+        loop_zone_list_w_area_dict = data["loop_zone_list_w_area_dict"]
+        return loop_or_piping in loop_zone_list_w_area_dict.keys()
+
+    class PumpRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section21Rule10.PumpRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False)
+            )
+
+        def get_calc_vals(self, context, data=None):
+            pump_b = context.baseline
+            # list_filter has checked the data
```

Comment:
Can you expand on this comment? Checked the data for what?

Reply:
My original intention was that the `loop_or_piping` data is checked in the previous level filter function.
`loop_or_piping` is a required data in the schema so we don't need this comment anymore.